### PR TITLE
[Sema] Report public imports of private module as error in asserts mode

### PR DESCRIPTION
**Pull request by xymus**
When running in assert mode make it an error for a public module to publicly import a private module.

### CODE REVIEW

**Comment by xymus** [comment]
@swift-ci Please smoke test###

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Comment by xymus** [comment]
@swift-ci Please smoke test Linux

File: test/Sema/implementation-only-import-suggestion.swift [hunk 1]
```diff
@@ -34,11 +34,11 @@
...
 #if PUBLIC_IMPORTS
 import PublicSwift
-import PrivateSwift // expected-warning{{private module 'PrivateSwift' is imported publicly from the public module 'main'}}
+import PrivateSwift // expected-error{{private module 'PrivateSwift' is imported publicly from the public module 'main'}}
 
 import PublicClang
-import PublicClang_Private // expected-warning{{private module 'PublicClang_Private' is imported publicly from the public module 'main'}}
-import FullyPrivateClang // expected-warning{{private module 'FullyPrivateClang' is imported publicly from the public module 'main'}}
+import PublicClang_Private // expected-error{{private module 'PublicClang_Private' is imported publicly from the public module 'main'}}
+import FullyPrivateClang // expected-error{{private module 'FullyPrivateClang' is imported publicly from the public module 'main'}}
```
Comment (hamishknight) [inline test/Sema/implementation-only-import-suggestion.swift:41]: I *think* you may need to mark this test as `REQUIRES: asserts` to make sure it doesn't fail on the no-asserts bots

Comment (xymus) [reply test/Sema/implementation-only-import-suggestion.swift:41]: That makes sense, thank you!

**Comment by xymus** [comment]
@swift-ci Please smoke test

**Pull request merged**